_doc-build-linux/action.yml

```diff
@@ -185,56 +185,112 @@ runs:
         sudo apt-get update
         sudo apt-get install -y poppler-utils
 
-    - name: "Check if requirements.txt file exists"
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Determine context.
+
+    - name: "Determine GitHub environment variables"
       shell: bash
       run: |
-        echo "EXISTS_DOC_REQUIREMENTS=$(if [ -f ${{ inputs.requirements-file }} ]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
+        if [[ -f "pyproject.toml" ]] && grep -q 'build-backend = "poetry\.core\.masonry\.api"' "pyproject.toml"; then
```

Comment:
Glad you also took into consideration setup.py projects (or even no-package doc projects using raw reqs files).